include/zephyr/dsp/utils.h

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (C) 2024 OWL Services LLC. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file zephyr/dsp/utils.h
+ *
+ * @brief Extra functions and macros for DSP
+ */
+
+#ifndef INCLUDE_ZEPHYR_DSP_UTILS_H_
+#define INCLUDE_ZEPHYR_DSP_UTILS_H_
+
+#include <stdint.h>
+#include <zephyr/kernel.h>
+#include <zephyr/dsp/dsp.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @ingroup math_dsp
+ * @defgroup math_dsp_utils_shifts Float/Fixed point shift conversion functions
+ */
+
+/**
+ * @ingroup math_dsp_utils_shifts
+ * @addtogroup math_dsp_basic_conv_to_float Fixed to Float point conversions
+ *
+ * Convert number Q7/Q15/Q31 to Float or Double representation with shift.
+ *
+ * There are separate functions for floating-point, Q7, Q15, and Q31 data types.
+ * @{
+ */
+
+/**
+ * @brief Convert a Q7 fixed-point value to a floating-point (float32_t) value with a left shift.
+ *
+ * @param src The input Q7 fixed-point value.
+ * @param n   The number of bits to left shift the input value (0 to 7).
```

Comment:
Here and below, documentation is for `n` but arg is `m`.

Reply:
Thanks! I will fix this in the next push.